File: .claude/worktrees/clever-torvalds/backend/mail_service.py
import os
import logging
from datetime import datetime

# NOTE: 生產環境建議使用真正的 SMTP 伺服器，例如 Gmail 或 SendGrid
# 此處實作與 Node.js 版對應的 Mock 邏輯

logger = logging.getLogger(__name__)

async def send_email_notification(to: str, title: str, doc_type: str, due_date: str):
    """
    發送郵件通知
    NOTE: 目前為 Mock 實作，僅記錄日誌。
    """
    logger.info(f"[MailService] Sending notification to {to} for {doc_type}: {title}")
    
    # 建構郵件內容 (與 Node.js 版本的模板一致)
    subject = f"【提醒】案件即將到期：{title}"
    body = f"""
您好，

系統提醒您，以下案件預計將在 3 天後 ({due_date}) 到期，請盡快處理：

案件類型：{doc_type}
案件標題/編號：{title}

祝 工作順利
Qualitas 工程管理系統
"""
    
    # 在此處可以加入 smtplib 的實作
    # try:
    #     import smtplib
    #     from email.mime.text import MIMEText
    #     msg = MIMEText(body)
    #     msg['Subject'] = subject
    #     msg['From'] = "noreply@qualitas-erp.com"
    #     msg['To'] = to
    #     # with smtplib.SMTP(host, port) as server:
    #     #     server.login(user, password)
    #     #     server.send_message(msg)
    # except Exception as e:
    #     logger.error(f"[MailService] Failed to send email: {e}")
    #     return False

    logger.info(f"[MailService] SUCCESS: Mock email sent to {to}")
    return True

File: .claude/worktrees/clever-torvalds/backend/mailService.js
const nodemailer = require('nodemailer');

// NOTE: In a real production environment, use environment variables for credentials
// const transportConfig = {
//   host: process.env.SMTP_HOST,
//   port: process.env.SMTP_PORT,
//   auth: {
//     user: process.env.SMTP_USER,
//     pass: process.env.SMTP_PASS
//   }
// };

// For demonstration, we use a mock transporter or a placeholder
const transporter = nodemailer.createTransport({
    host: 'smtp.ethereal.email',
    port: 587,
    auth: {
        user: 'mock_user@ethereal.email',
        pass: 'mock_pass'
    }
});

async function sendEmailNotification(to, title, docType, dueDate) {
    console.log(`[MailService] Sending notification to ${to} for ${docType}: ${title}`);

    const mailOptions = {
        from: '"Qualitas System" <noreply@qualitas-erp.com>',
        to: to,
        subject: `【提醒】案件即將到期：${title}`,
        text: `
您好，

系統提醒您，以下案件預計將在 3 天後 (${dueDate}) 到期，請盡快處理：

案件類型：${docType}
案件標題/編號：${title}

祝 工作順利
Qualitas 工程管理系統
        `,
        html: `
            <div style="font-family: sans-serif; padding: 20px; border: 1px solid #eee; border-radius: 8px;">
                <h2 style="color: #f59e0b;">⚠️ 案收到期提醒</h2>
                <p>您好，</p>
                <p>系統提醒您，以下案件預計將在 <strong>3 天後 (${dueDate})</strong> 到期，請盡快處理：</p>
                <table style="width: 100%; border-collapse: collapse; margin: 20px 0;">
                    <tr>
                        <td style="padding: 10px; border: 1px solid #ddd; background: #f9f9f9; width: 120px;"><strong>案件類型</strong></td>
                        <td style="padding: 10px; border: 1px solid #ddd;">${docType}</td>
                    </tr>
                    <tr>
                        <td style="padding: 10px; border: 1px solid #ddd; background: #f9f9f9;"><strong>案件標題/編號</strong></td>
                        <td style="padding: 10px; border: 1px solid #ddd;">${title}</td>
                    </tr>
                </table>
                <p style="color: #666; font-size: 14px;">此為系統自動發送，請勿直接回覆。</p>
                <hr style="border: none; border-top: 1px solid #eee; margin: 20px 0;">
                <p style="font-weight: bold; color: #2563eb;">Qualitas 工程管理系統</p>
            </div>
        `
    };

    try {
        // In a real scenario, this would send the email
        // const info = await transporter.sendMail(mailOptions);
        // console.log('[MailService] Message sent: %s', info.messageId);

        // Mock success for now
        console.log(`[MailService] SUCCESS: Email sent to ${to}`);
        return true;
    } catch (error) {
        console.error('[MailService] Error sending email:', error);
        return false;
    }
}

module.exports = {
    sendEmailNotification
};
